src/pages/suggestions/edit/[suggestion].tsx

```diff
@@ -0,0 +1,193 @@
+import React from "react"
+import SuggestionPageSection from "@/components/new-suggestion/suggestion-page-section"
+import { GetServerSideProps } from "next"
+import { createPagesServerClient } from "@supabase/auth-helpers-nextjs"
+import {
+  deleteSuggestionInstruments,
+  getSuggestion,
+  updateSuggestion,
+  updateSuggestionInstruments,
+} from "@/services/suggestion.service"
+import { Suggestion } from "@/types/database-types"
+import { InputsSongInformation, NewSuggestionInstrument } from "@/interfaces/new-suggestion"
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { useRouter } from "next/router"
+import { Area } from "@/constants/area"
+import { AppState } from "@/redux/store"
+import { useDispatch, useSelector } from "react-redux"
+import { mapEditInstruments } from "@/helpers/new-suggestion.helper"
+import {
+  updateEditSuggestion,
+  updateLastDeletedInstrumentUuid,
+  updateLastEditedUuid,
+} from "@/redux/slices/edit-suggestion.slice"
+import { setActiveArea } from "@/redux/slices/new-suggestion.slice"
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const supabase = createPagesServerClient(context)
+  // Check if we have a session
+  const {
+    data: { session },
+  } = await supabase.auth.getSession()
+  const { params } = context
+
+  try {
+    // Check if the suggestion exists
+    const { data } = await getSuggestion(supabase, params?.suggestion as string)
+    if (data == null) return { notFound: true }
+
+    // If the suggestion has no id
+    const author: { id: string } = data?.author as { id: string }
+    if (author?.id === undefined) return { notFound: true }
+
+    // If the user does not match with the author of the suggestion
+    if (session?.user.id !== author.id)
+      return {
+        redirect: {
+          destination: "/403",
+          permanent: false,
+        },
+      }
+
+    return { props: { suggestion: data } }
+  } catch {
+    return { notFound: true }
+  }
+}
+
+interface EditSuggestionPageProps {
+  suggestion: Suggestion
+}
+
+const EditSuggestionPage = (props: EditSuggestionPageProps) => {
+  const user = useUser()
+  const supabase = useSupabaseClient<Database>()
+  const router = useRouter()
+  const dispatch = useDispatch()
+
+  const newSuggestion = useSelector((state: AppState) => state.editSuggestion)
```

Comment:
Since the only thing you retrieve from this is the uuid, maybe just do 
`const lastEditedUuid = useSelector((state: AppState) => state.editSuggestion.lastEditedUuid)`